tokio-threadpool/src/pool/mod.rs

```diff
@@ -450,3 +448,31 @@ impl PartialEq for Pool {
 
 unsafe impl Send for Pool {}
 unsafe impl Sync for Pool {}
+
+// Return a thread-specific, 32-bit, non-zero seed value suitable for a 32-bit
+// PRNG. This uses one libstd RandomState for a default hasher and hashes on
+// the current thread ID to obtain an unpredictable, collistion resitant seed.
+fn prng_seed() -> u32 {
+    // This obtains a small number of random bytes from the host system (for
+    // example, on unix via getrandom(2)) in order to seed an unpredictable and
+    // HashDoS resistant 64-bit hash function (currently: `SipHasher13` with
+    // 128-bit state). We only need one of these, to make the seeds for all
+    // process threads different via hashed IDs, collision resistant, and
+    // unpredictable.
+    lazy_static! {
```

Comment:
I probably wouldn't bother with a `lazy_static`. Instead, why not create one when the pool is created and pass the seed in to each worker when that is initialized?

Reply:
I actually tried that first, but get compile errors because `&self ` is not constant (in `fn rand_usize`) for the thread local PRNG.  It is an appropriate compiler lint, I think, because if its thread local then it needs to be the same across all pools... I mean, to the extent that the same thread could even be a part of multiple pools (a detail the compiler apparently doesn't know).

Reply:
Ok, not critical for merging this PR. We can follow up if we care.